Sources/LanguageServerProtocol/Command.swift

```diff
@@ -0,0 +1,133 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SKSupport
+
+/// A `CommandDataType` represents the underlying data of a custom `Command`.
+public protocol CommandDataType: Codable {
```

Comment:
I think it's more accurate to describe this as "A `CommandDataType` represents the arguments required to execute a `Command`."

Also, If you agree with my suggestion to keep "arguments", I'll rename it as `CommandArgsType`.

Reply:
I don't know whether requiring all these three properties may be very restrictive to the model or not. I imagine a hypothetic future "Organize imports" command action that would only require a reference to the `textDocument` to do its job. Also, I'd copy here the property docs you have in `SemanticRefactorCommandDataType`.